package.json

```diff
@@ -8,18 +8,19 @@
   "scripts": {
     "lerna:version": "lerna version",
     "lerna:publish": "lerna publish from-git --yes",
-    "ci": "npm-run-all build lint format:validate test",
+    "ci": "npm-run-all build lint format:fix-generated format:validate test",
     "test": "lerna run test --stream",
     "lint": "eslint packages/**/*.js",
     "format:fix": "prettier --write \"**/*.@(js|json|md)\"",
     "format:validate": "prettier --list-different \"**/*.@(js|json|md)\"",
-    "build": "node packages/java-parser/scripts/unicode.js packages/java-parser/resources/Unicode/UnicodeData.txt; prettier --write packages/java-parser/src/unicodesets.js; node packages/java-parser/scripts/gen-diagrams.js",
+    "format:fix-generated": "prettier --write packages/java-parser/src/unicodesets.js packages/java-parser/src/gen/grammar.json;",
+    "build": "node packages/java-parser/scripts/unicode.js packages/java-parser/resources/Unicode/UnicodeData.txt; node packages/java-parser/scripts/gen-diagrams.js",
     "prettier": "prettier --plugin=./packages/prettier-plugin-java --parser=java",
     "update-test-outputs": "node packages/prettier-plugin-java/scripts/update-test-outputs.js"
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+      "pre-commit": "lint-staged && node packages/java-parser/scripts/gen-grammar.js && git add packages/java-parser/src/gen/grammar.json"
```

Comment:
I am thinking that maybe we should not commit the generated serialized grammar json.
What do you think?

Reply:
We probably want a script to generate the serialized format in package json
and call it in the hook as "npm run SCRIPT_NAME"

Reply:
I think we should stay consistent, we are already committing a generated file (unicodesets.js) and using a script to maintain every once in a while.
Imo, we should commit the grammar.json file and use the script to serialize the grammar to maintain. I'm unsure whether we shoud regenerate it. It generates a new grammar at each pre-commit but I don't think it adds it to the commit at the moment: right now it's mostly used for local development.